### PR TITLE
Adding tests for container index CI

### PR DESCRIPTION
**Pull request by mohammedzee1000**
- Setting up basic test framework
- Adding tests for schema validation.

### CODE REVIEW

**Comment by mohammedzee1000** [comment]
Part of https://github.com/CentOS/container-pipeline-service/issues/583

**Comment by mohammedzee1000** [comment]
#dotests failure not on PR code

**Comment by mohammedzee1000** [comment]
#dotests

File: ci/container_index/lib/checks/schema_validation.py [hunk 1]
```diff
@@ -135,12 +135,12 @@ def _perform_validation(self):
         if (not isinstance(
                 self.validation_data.get(FieldKeys.PREBUILD_SCRIPT), str)):
             self._invalidate("prebuild-script must be a valid string.")
+        elif len(self.validation_data.get(FieldKeys.PREBUILD_SCRIPT)) <= 0:
```
Comment (navidshaikh) [inline ci/container_index/lib/checks/schema_validation.py]: how come length be of `<  0` ?  `== 0` is just fine.

File: ci/tests/test_00_unit/test_05_index_ci/indexcibase.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    node = "controller"
+
+    @staticmethod
+    def _print_init_msg(msg):
+        count = len(msg)
+        print("\n")
+        print("=" * count)
+        print(msg)
+        print("=" * count)
```
Comment (navidshaikh) [inline ci/tests/test_00_unit/test_05_index_ci/indexcibase.py]: unnecessary 4 lines of print statements which could have been reduced to
`\n{}\n{}\n{}`

File: ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_06_git_url_validation/__init__.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        self.assertTrue(
+            schema_validation.GitURLValidator(
+                {
+                    FieldKeys.GIT_URL: "https://github.com/test/test"
+                },
+                DUMMY_INDEX_FILE
+            ).validate().success
+        )
+
+    def test_02_validation_fails_missing_job_id(self):
```
Comment (navidshaikh) [inline ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_06_git_url_validation/__init__.py]: I think the method name is left unchanged?

File: ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_06_git_url_validation/__init__.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        self.assertFalse(
+            schema_validation.GitURLValidator(
+                {
+                    "Git-URL": "https://github.com/test/test"
+                },
+                DUMMY_INDEX_FILE
+            ).validate().success
+        )
+
+    def test_03_validation_fails_job_id_not_string(self):
```
Comment (navidshaikh) [inline ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_06_git_url_validation/__init__.py]: I think the method name is left unchanged?

File: ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_07_git_path_validation/__init__.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        self.assertTrue(
+            schema_validation.GitPathValidator(
+                {
+                    FieldKeys.GIT_PATH: "/"
+                },
+                DUMMY_INDEX_FILE
+            ).validate().success
+        )
+
+    def test_02_validation_fails_missing_job_id(self):
```
Comment (navidshaikh) [inline ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_07_git_path_validation/__init__.py]: I think the method name is left unchanged?

File: ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_07_git_path_validation/__init__.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        self.assertFalse(
+            schema_validation.GitPathValidator(
+                {
+                    "Git-Path": "/"
+                },
+                DUMMY_INDEX_FILE
+            ).validate().success
+        )
+
+    def test_03_validation_fails_job_id_not_string(self):
```
Comment (navidshaikh) [inline ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_07_git_path_validation/__init__.py]: I think the method name is left unchanged?

File: ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_10_notify_email_validation/__init__.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        self.assertFalse(
+            schema_validation.NotifyEmailValidator(
+                {
+                    "Notify-Email": "test@example.com"
+                },
+                DUMMY_INDEX_FILE
+            ).validate().success
+        )
+
+    def test_03_validation_fails_notify_email_not_string(self):
```
Comment (navidshaikh) [inline ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_10_notify_email_validation/__init__.py]: are we just checking for string or verifying using email pattern regex ?

Comment (mohammedzee1000) [reply ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_10_notify_email_validation/__init__.py]: atm im just checking for string. I am still trying to find an email regex in python that accepts everythong

File: ci/tests/test_00_unit/test_05_index_ci/indexcibase.py (hunk 2 above)
Comment (mohammedzee1000) [reply ci/tests/test_00_unit/test_05_index_ci/indexcibase.py]: Removed

File: ci/container_index/lib/checks/schema_validation.py (hunk 1 above)
Comment (mohammedzee1000) [reply ci/container_index/lib/checks/schema_validation.py]: Sorted

File: ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_06_git_url_validation/__init__.py (hunk 3 above)
Comment (mohammedzee1000) [reply ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_06_git_url_validation/__init__.py]: Fixed

File: ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_06_git_url_validation/__init__.py (hunk 4 above)
Comment (mohammedzee1000) [reply ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_06_git_url_validation/__init__.py]: Fixed

File: ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_07_git_path_validation/__init__.py (hunk 5 above)
Comment (mohammedzee1000) [reply ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_07_git_path_validation/__init__.py]: Fixed

File: ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_07_git_path_validation/__init__.py (hunk 6 above)
Comment (mohammedzee1000) [reply ci/tests/test_00_unit/test_05_index_ci/test_00_schema_validation/test_07_git_path_validation/__init__.py]: Fixed

**Comment by navidshaikh** [comment]
#dotests

**Pull request merged**